docs/tutorials/Manual-Signature-Verification/Readme.md

```diff
@@ -0,0 +1,268 @@
+
+We have a special `PowerAuthRequestCanonizationUtils` class ([link](https://github.com/wultra/powerauth-crypto/blob/develop/powerauth-java-http/src/main/java/io/getlime/security/powerauth/http/PowerAuthRequestCanonizationUtils.java)) for this task. The code is a bit too long and boring for a copy-paste to this tutorial. If you need to handle signed `GET` requests, have a look into our implementation.
+
+## Building the Signature Base String
+
+Now, when we have the request data and parsed HTTP header, we may proceed to building the signature base string. This is the string value that was actually signed on the mobile device.
+
+Building the signature base string is simple once we have all the data we talked about earlier in place. We use our own `PowerAuthHttpBody` class ([link](https://github.com/wultra/powerauth-crypto/blob/develop/powerauth-java-http/src/main/java/io/getlime/security/powerauth/http/PowerAuthHttpBody.java)) to do the most of the logic.
+
+The important Java code is the following:
+
+```java
+public static String getSignatureBaseString(String httpMethod, String requestUri, byte[] nonce, byte[] data) {
+
+    String requestUriHash = "";
+    if (requestUri != null) {
+        byte[] bytes = requestUri.getBytes(StandardCharsets.UTF_8);
+        requestUriHash = BaseEncoding.base64().encode(bytes);
+    }
+
+    String dataBase64 = "";
+    if (data != null) {
+        dataBase64 = BaseEncoding.base64().encode(data);
+    }
+
+    String nonceBase64 = "";
+    if (nonce != null) {
+        nonceBase64 = BaseEncoding.base64().encode(nonce);
+    }
+
+    return (httpMethod != null ? httpMethod.toUpperCase() : "GET")
+            + "&" + requestUriHash
+            + "&" + nonceBase64
+            + "&" + dataBase64;
+}
+```
+
+To call the method, you need to know:
+
+- HTTP request method - Comes with the HTTP request.
+- The "request URI" value - The request URI ID, a constant that we talked about earlier and that client and server need to share beforehand.
+    - **Note: This is not the absolute HTTP URI path, only a constant that usually resembles it, such as `/login`.**
+- The "nonce" - Comes in the HTTP header.
+    - **Note: The nonce value should be Base64 encoded only once. In our code, we receive nonce in the HTTP header as a Base64 encoded string, we decode it to the bytes, and when building a signature base string, we encode it to Base64 again. While this looks like extra work, at least we can validate the correct format of the value as well as sufficient length of the underlying `byte[]`.**
+- The request data - We obtained those in the section above.
+
+The result should look something like this:
+
+```
+POST&L29wZXJhdGlvbi9hdXRob3JpemU=&j1MADdlwDmN3ZV7cFt74Qg==&eyJyZXF1ZXN0T2JqZWN0I
+jp7ImlkIjoiNzBkMDM5MjktNmZkZC00MzE1LTk1NzQtYzk3ZGM2ZDU2YWJhIiwiZGF0YSI6IkEyIn19&
+Ec1RlAr6B3Il6wEg9OQLXA==
+```
+
+At this moment, we should have everything to call the PowerAuth Server API.
```

Comment:
What about `applicationSecret`, isn't it missing?

Reply:
The secret is between the app and the server, intermediate systems do not have it...